modules/custom/activity_creator/src/ActivityNotifications.php

```diff
@@ -82,38 +88,58 @@ public function getNotificationsActivities(AccountInterface $account, array $sta
    * @param \Drupal\Core\Session\AccountInterface $account
    *   Account object.
    *
-   * @return int
-   *   Number of remaining notifications.
+   * @return bool
+   *   TRUE or FALSE depending upon update status.
    */
-  public function markAllNotificationsAsSeen(AccountInterface $account): int {
+  public function markAllNotificationsAsSeen(AccountInterface $account): bool {
     // Retrieve all the activities referring this entity for this account.
-    $ids = $this->getNotificationIds($account);
-    $this->changeStatusOfActivity($ids, ACTIVITY_STATUS_SEEN);
+    if (!empty($ids = $this->getNotificationIds($account, [ACTIVITY_STATUS_RECEIVED]))) {
+      return $this->changeStatusOfActivity($ids, $account, ACTIVITY_STATUS_SEEN);
+    }
 
-    return 0;
+    // Log as notice.
+    $this->loggerFactory->get('activity_creator')->notice('There are no notification to be marked as seen for user @id', [
+      '@id' => $account->id(),
+    ]);
```

Comment:
@bramtenhove I update the service to be completely independent as a service should literally be as it should just manipulate information or provide us with the information we asked for.

changeStatusOfActivity() will return TRUE or FALSE depending upon the execution of update query. 

In this case, markAllNotificationsAsSeen()  is returning whatever changeStatusOfActivity() returns but it also checks ids() for being empty. So, is it a good idea to log a notice and return FALSE or just return FALSE after if statement.